app/controllers/shopify_app/sessions_controller.rb

```diff
@@ -46,6 +45,7 @@ def login_shop
       sess = ShopifyAPI::Session.new(shop_name, token)
       session[:shopify] = ShopifyApp::SessionRepository.store(sess)
       session[:shopify_domain] = shop_name
+      session[:shopify_user] = associated_user if associated_user.present?
```

Comment:
I think this should be passed to `ShopifyAPI::Session.new` as [its 3rd argument](http://www.rubydoc.info/gems/shopify_api/ShopifyAPI%2FSession:initialize). This way, one can persist email, locale, etc... when storing a session.